### PR TITLE
asserts: require components in validation sets to have revisions if their associated snap has one

### PR DESCRIPTION
**Pull request by andrewphelpsj**
We're going to require this for now, might be lifted later. I also added some more context to error messages while in here.

### CODE REVIEW

File: asserts/validation_set.go [hunk 1]
```diff
@@ -130,21 +130,21 @@ func checkValidationSetSnap(snap map[string]interface{}) (*ValidationSetSnap, er
...
+		Name:       snapName,
 		SnapID:     snapID,
 		Presence:   presence,
 		Revision:   snapRevision,
 		Components: components,
 	}, nil
 }
 
-func checkValidationSetComponents(snap map[string]interface{}, what string) (map[string]ValidationSetComponent, error) {
+func checkValidationSetComponents(snap map[string]interface{}, snapRevision int, snapName string, what string) (map[string]ValidationSetComponent, error) {
```
Comment (pedronis) [inline asserts/validation_set.go:147]: unrelated but it's actually unclear that "what" ends up actually used in errors, maybe we can leave it out and pass snapName first to the function?

File: asserts/validation_set.go [hunk 2]
```diff
@@ -176,12 +176,12 @@ func checkValidationSetComponents(snap map[string]interface{}, what string) (map
 	return components, nil
 }
 
-func checkValidationSetComponent(comp map[string]interface{}, name string) (ValidationSetComponent, error) {
-	if err := naming.ValidateSnap(name); err != nil {
-		return ValidationSetComponent{}, fmt.Errorf("invalid component name %q", name)
+func checkValidationSetComponent(comp map[string]interface{}, compName string, snapRevision int, snapName string) (ValidationSetComponent, error) {
```
Comment (pedronis) [inline asserts/validation_set.go:179]: nitpick: args order:  compName, comp, snapName, snapRevision

**Review by pedronis — APPROVED** [review]
couple nitpicks

**Review by MiguelPires — APPROVED** [review]
lgtm

**Pull request merged**